velox/docs/programming-guide/chapter01.rst

```diff
@@ -0,0 +1,376 @@
+    }
+
+    > 0: true
+    > 1: false
+    > 2: true
+    > 3: false
+
+We can also use printNulls helper function to print the null flags:
+
+.. code-block:: c++
+
+    LOG(INFO) << printNulls(nulls, 10);
+
+    > 99 out of 104 rows are null: n.n.n.n.n.
+
+    LOG(INFO) << printNulls(nulls);
+
+    > 99 out of 104 rows are null: n.n.n.n.n.nnnnnnnnnnnnnnnnnnnn
+
+printNulls function takes a nulls buffer and an optional maxBitsToPrint number
+which has a default value of 30.
+
+.. code-block:: c++
+
+    std::string printNulls(
+       const BufferPtr& nulls,
+       vector_size_t maxBitsToPrint = 30);
+
+This function returns a string where each character represents a single null
+flag: ‘n’ for null and ‘.’ for non-null. The result string also includes a
+summary prefix telling us how many entries are null. Notice that the summary
+says that there are a total of 104 entries, not 100. This is because BufferPtr
+doesn’t know that it is used to store null flags. It just knows its size in
+bytes, which is 13. The last byte has some bits unused.
+
+We have learned how to allocate memory and fill it in with values and null
+flags. We are now ready to assemble a flat vector to hold data for a single
+column. Let’s make a vector to store 100 sequential BIGINT values with every
+other value being null: [0, null, 2, null, 4, null, 6,..].
+
+.. code-block:: c++
+
+    #include "velox/vector/FlatVector.h"
+
+    auto vector = std::make_shared<FlatVector<int64_t>>(
+       pool.get(), BIGINT(), nulls, 100, values, std::vector<BufferPtr>{});
+
+    LOG(INFO) << vector->toString();
+
+    > [FLAT BIGINT: 100 elements, 50 nulls]
+
+To make a vector, we use FlatVector<T> class template with T being int64_t
+(64-bit integer). We pass a pointer to MemoryPool, a Type object that describes
+the type of values to store, nulls buffer, number of values, values buffer and
+an empty list of string buffers. Let’s ignore the string buffers for now. We’ll
+discuss these later.
+
+FlatValue<T> class can be used to store values of primitive types. The following
+types are supported in Velox:
+
+======================  ===========================    ==================
+Type                    C++ Type                       Description
+======================  ===========================    ==================
+BOOLEAN                 bool                           A boolean flag: true or false.
+TINYINT                 int8_t                         8-bit integer.
+SMALLINT                int16_t                        16-bit integer.
+INTEGER                 int32_t	                       32-bit integer.
+BIGINT                  int64_t                        64-bit integer.
+REAL                    float                          32-bit floating point number.
+DOUBLE                  double                         64-bit floating point number.
+VARCHAR                 struct StringView              Variable width string.
```

Comment:
Would be worth adding some content on how to handle DATE, TIMESTAMP and DECIMALs to this list.